provider/cldr/src/cldr_serde/week_data.rs

```diff
@@ -0,0 +1,123 @@
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Serde structs representing CLDR JSON weekData.json files.
+//!
+//! Sample file:
+//! https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/weekData.json
+
+use core::convert::TryFrom;
+use serde::{Deserialize, Deserializer};
+use std::collections::BTreeMap;
+use std::num::ParseIntError;
+use std::str::FromStr;
+use tinystr::{tinystr4, TinyStr4};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum Weekday {
+    Mon,
+    Tue,
+    Wed,
+    Thu,
+    Fri,
+    Sat,
+    Sun,
+}
+
+impl From<&Weekday> for icu_calendar::types::IsoWeekday {
+    fn from(day: &Weekday) -> Self {
+        use icu_calendar::types::IsoWeekday;
+        match day {
+            Weekday::Mon => IsoWeekday::Monday,
+            Weekday::Tue => IsoWeekday::Tuesday,
+            Weekday::Wed => IsoWeekday::Wednesday,
+            Weekday::Thu => IsoWeekday::Thursday,
+            Weekday::Fri => IsoWeekday::Friday,
+            Weekday::Sat => IsoWeekday::Saturday,
+            Weekday::Sun => IsoWeekday::Sunday,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Territory {
+    Region(TinyStr4),
+    AltVariantRegion(TinyStr4),
+}
+
+/// The string used to represent the default territory.
+pub const DEFAULT_TERRITORY: Territory = Territory::Region(tinystr4!("001"));
+
+/// Suffix used to denote alternative week data variants for a given territory (e.g. English BC/AD v English BCE/CE).
+const ALT_VARIANT_SUFFIX: &str = "-alt-variant";
+
+impl<'de> Deserialize<'de> for Territory {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct TerritoryVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for TerritoryVisitor {
+            type Value = Territory;
+
+            fn expecting(&self, formatter: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                write!(
+                    formatter,
+                    "a valid Unicode Language Identifier or default territory literal"
+                )
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                if let Some(prefix) = s.strip_suffix(ALT_VARIANT_SUFFIX) {
+                    return Ok(Territory::AltVariantRegion(
+                        prefix
+                            .parse::<TinyStr4>()
+                            .map_err(serde::de::Error::custom)?,
+                    ));
+                }
+
+                Ok(Territory::Region(
+                    s.parse::<TinyStr4>().map_err(serde::de::Error::custom)?,
+                ))
+            }
+        }
+
+        deserializer.deserialize_string(TerritoryVisitor)
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(try_from = "String")]
+pub struct U8(pub u8);
```

Comment:
question: What's the benefit of wrapping this in another type?

Reply:
Ah, serde json doesn't natively support deserialising strings to numbers. As the value here is used as a map value annotations can't AFAICT be used to coerce it (https://github.com/serde-rs/json/issues/412).